File: engine/src/main/java/org/epdb/engine/dto/StringValue.java
package org.epdb.engine.dto;

public record StringValue(String value) implements ColumnValue {

    @Override
    public int comparesTo(ColumnValue other) {
        if(other instanceof StringValue(String otherString)) {
            return this.value.compareTo(otherString);
        }

        throw new ClassCastException("Cannot compare StringValue with " + other.getClass().getSimpleName());
    }
}

File: modules/engine/src/main/kotlin/org/epdb/engine/columntypes/StringValue.kt
package org.epdb.engine.columntypes

data class StringValue(val value: String) : ColumnValue {

    override fun comparesTo(other: ColumnValue): Int {
        return when (other) {
            is StringValue -> {
                this.value.compareTo(other.value)
            }

            else -> throw ClassCastException("Cannot compare IntValue with ${other.javaClass.getSimpleName()}")
        }
    }
}